publishing/datasetAPI/get_observation_test.go

```diff
@@ -89,19 +90,73 @@ func TestSuccessfullyGetObservationForVersion(t *testing.T) {
 				response.Value("links").Object().Value("self").Object().Value("href").String().Match(".+/datasets/" + datasetID + "/editions/" + edition + "/versions/2/observations\\?aggregate=cpi1dim1S40403&geography=K02000001&time=Aug-16$")
+				response.Value("links").Object().Value("version").Object().Value("href").String().Match("/datasets/" + datasetID + "/editions/" + edition + "/versions/2$")
 				response.Value("links").Object().Value("version").Object().Value("id").Equal("2")
+				response.Value("observations").Array().Length().Equal(1)
+				response.Value("observations").Array().Element(0).Object().Value("observation").Equal("154.6")
+				response.Value("offset").Equal(0)
+				response.Value("total_observations").Equal(1)
+				response.Value("unit_of_measure").Equal("Pounds Sterling")
+			})
+		})
+
+		Convey("When a request is made to get an observations resource containing more than one observation for an unpublished version", func() {
+			Convey("Then the response body contains the expected observations data", func() {
+				response := datasetAPI.GET("/datasets/{id}/editions/{edition}/versions/2/observations", datasetID, edition).
+					WithQueryString("time=Aug-16&geography=K02000001&aggregate=*").
+					WithHeader(florenceTokenName, florenceToken).
+					Expect().Status(http.StatusOK).JSON().Object()
+
+				response.Value("dimensions").Object().Value("geography").Object().Value("option").Object().Value("href").String().Match("/codelists/708064B3-A808-449B-9041-EA3A2F72CFAF/codes/K02000001$")
+				response.Value("dimensions").Object().Value("geography").Object().Value("option").Object().Value("id").Equal("K02000001")
+				response.Value("dimensions").Object().Value("time").Object().Value("option").Object().Value("href").String().Match("/codelists/608064B3-A808-449B-9041-EA3A2F72CFAE/codes/Aug-16$")
+				response.Value("dimensions").Object().Value("time").Object().Value("option").Object().Value("id").Equal("Aug-16")
+				response.Value("limit").Equal(10000)
+				response.Value("links").Object().Value("dataset_metadata").Object().Value("href").String().Match("/datasets/" + datasetID + "/editions/" + edition + "/versions/2/metadata$")
+				response.Value("links").Object().Value("self").Object().Value("href").String().Match(".+/datasets/" + datasetID + "/editions/" + edition + "/versions/2/observations\\?aggregate=\\%2A&geography=K02000001&time=Aug-16$")
+				response.Value("links").Object().Value("version").Object().Value("href").String().Match("/datasets/" + datasetID + "/editions/" + edition + "/versions/2$")
+				response.Value("links").Object().Value("version").Object().Value("id").Equal("2")
+				response.Value("observations").Array().Length().Equal(137)
+
+				// check two observations in observations array
+				for i := 0; i < len(response.Value("observations").Array().Iter()); i++ {
```

Comment:
this `for` loop is a tad broken...
1. use `for _, observation := range response.Value("observations").Array().Iter() {`
2. put `count := 0` _outside_ the loop
3. test that `count` is 2 *after* the loop is done

Reply:
I will use the range for loop and remove counter as it appears to be eating up time then gaining time.